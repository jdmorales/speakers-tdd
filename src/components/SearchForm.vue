  <template>
  <div>
    <input type="text" v-model="inputData">
    <button @click="fetchSpeaker(inputData) "></button>
  </div>
</template>

<script>
import { mapActions } from 'vuex'

export default {
  data() {
    return {
      inputData: ''
    }
  },
  methods: {
    ...mapActions({
      fetchSpeaker: 'FETCH_SPEAKER'
    })
  }
}
</script>

<style>

</style>