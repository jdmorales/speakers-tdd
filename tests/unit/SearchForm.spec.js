import { shallowMount } from '@vue/test-utils'
import SearchForm from '@/components/SearchForm'

describe('SearchForm', () => {
  
  const build = () => {
    const wrapper = shallowMount(SearchForm)

    return {
      wrapper,
      input: () => wrapper.find('input'),
      button: () => wrapper.find('button')
    }
  }

  it('it renders the component', () => {
    // arrange
    const { wrapper } = build()
    // assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('it renders the right', () => {
    // arrange
    const { input, button } = build()
    // assert
    expect(input().exists()).toBe(true)
    expect(button().exists()).toBe(true)
  })

  it('binds the input data', () => {
    // arrange
    const { wrapper, input } = build()
    // Set local data to Speaker
    wrapper.setData({
      inputData: 'Speaker'
    })
    // assert
    expect(input().element.value).toBe('Speaker')
  })
})
