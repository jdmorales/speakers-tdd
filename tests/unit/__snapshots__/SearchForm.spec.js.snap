// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm it renders the component 1`] = `<div><input type="text"> <button></button></div>`;
